server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/util/LocalityMappingUtils.java

```diff
@@ -184,4 +185,35 @@ private boolean tokenMatches(String token, String position) {
     return trimmedToken.equals(positionText)
         || trimmedToken.equals(positionText.replace("<EOF>", ""));
   }
+
+  /**
+   * Use this method when a token is not found the the Token to Locality mapping. This method tries
+   * to find the nearest locality for a provided token.
+   *
+   * For e.g. A token at line position 0, char start index 3, char end index 6 is passed( value - TES).
+   * But the same token is not present in the mapping.
+   * Then this method tries to find the nearest relevant token. And matches to token
+   * TEST, which is at position 0, and char start index at 3 and char end index is 7
+   *
+   * @param referenceToken tokens of type {@link org.antlr.v4.runtime.CommonToken} provided by
+   *     {@link CobolLexer}
+   * @param mapping A Map of Token to Locality for a document in analysis.
+   * @return Locality for a passed token.
+   */
+  public Optional<Locality> getNearestLocality(
+      Token referenceToken, @NonNull Map<Token, Locality> mapping) {
+    if (Objects.isNull(referenceToken)) return Optional.empty();
+    return mapping.entrySet().stream()
+        .filter(
+            entry ->
+                StringUtils.normalizeSpace(entry.getKey().getText())
+                        .contains(StringUtils.normalizeSpace(referenceToken.getText()))
```

Comment:
You can extract it out of the filter. Thus you normalizeSpace for token only once, you don't need to do it for each mapping entry.

Reply:
Done.